### ISSUE TITLE
GPIO: Don't rely on embedded-hal-traits

### ISSUE THREAD

**Issue by David-OConnor** [nrf-rs/nrf-hal#349]
Ie, remove `unwrap()` requirement when setting pins in both internal and external code.

**Comment by jonas-schievink** [comment]
Yeah, we can probably add inherent methods for controlling the pins. Note that there's also a pending redesign of the whole GPIO module in https://github.com/nrf-rs/nrf-hal/pull/301.

**Comment by David-OConnor** [comment]
Of note: This also applies to some parts of the timer and uarte modules.
